src/pydvl/influence/types.py

```diff
@@ -1,48 +1,53 @@
 from abc import ABC
-from typing import Callable, Iterable, Optional, Tuple
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
 
 from numpy import ndarray
 
-__all__ = [
-    "TwiceDifferentiable",
-    "MatrixVectorProduct",
-    "MatrixVectorProductInversionAlgorithm",
-]
+try:
+    import torch
+    import torch.nn as nn
+
+    _TORCH_INSTALLED = True
+except ImportError:
+    _TORCH_INSTALLED = False
+
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
+
+
+__all__ = ["TwiceDifferentiable", "TensorType"]
+
+TensorType = Union["NDArray", "torch.TensorType"]
```

Comment:
I didn't know about `torch.TensorType`. It seems it was added in [this commit](https://github.com/pytorch/pytorch/commits/9f3167e) and should be part of Pytorch 1.10.

We should either provide an alternative if it doesn't exist, limit pytorch version to those after 1.10 or use `torch.Tensor` instead.